### PR TITLE
Reset RocksDb when the metadata store fails to open

### PR DESCRIPTION
**Pull request by jbayardo**

### CODE REVIEW

File: Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs [hunk 1]
```diff
@@ -231,6 +231,9 @@ private struct VersionConstants
         /// <param name="openBulkLoad">
         /// Have RocksDb open for bulk loading.
         /// </param>
+        /// <param name="onStoreReset">
+        /// Callback for when the store gets reset due to `onFailureDeleteExistingStoreAndRetry`
```
Comment (kewilli) [inline Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs]: >onFailureDeleteExistingStoreAndRetry [](start = 59, length = 36)

Nit: `<paramref>`

File: Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs [hunk 2]
```diff
@@ -304,6 +308,9 @@ private struct VersionConstants
...
```
Comment (kewilli) [inline Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs]: >onFailureDeleteExistingStoreAndRetry [](start = 59, length = 36)

Ditto

**Comment by SergeyTeplyakov** [comment]
>             bool openReadOnly = false,

it seems that its time to create a configuration class:)

---
Refers to: Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs:243 in 4140640. [](commit_id = 41406400096fd1c8998a558ec9051b32bc4d06f7, deletion_comment = False)

**Pull request merged**